src/main/puzzle/piece/piece-manager.gd

```diff
@@ -80,15 +78,26 @@ func _physics_process(_delta: float) -> void:
 
 
 func clear_piece() -> void:
-	if $States.get_state() != $States/None:
-		$States.set_state($States/None)
 	active_piece = ActivePiece.new(PieceTypes.piece_null)
 	tile_map_dirty = true
 
 
+"""
+Writes the current piece to the playfield, checking whether it makes any boxes or clears any lines.
+
+Returns true if the newly written piece results in a pause of some sort.
+"""
+func write_piece_to_playfield() -> bool:
+	var result := playfield.write_piece(active_piece.pos, active_piece.orientation, active_piece.type,
+			piece_speed.line_clear_delay)
+	clear_piece()
+	return result
+
+
```

Comment:
This comment is inaccurate.

Reply:
The comment is accurate, Github was just displaying the surrounding code in a misleading way.